engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsSameObjectNode.java

```diff
@@ -23,6 +25,15 @@ public static IsSameObjectNode build() {
 
   public abstract boolean execute(@AcceptsError Object left, @AcceptsError Object right);
 
+  @Specialization
+  boolean isSameDouble(double left, double right) {
+    if (Double.isNaN(left) && Double.isNaN(right)) {
+      return true;
+    } else {
+      return left == right;
+    }
```

Comment:
```suggestion
    return (Double.isNaN(left) && Double.isNaN(right)) || (left == right);
```
personal preference feel free to ignore :p